### PR TITLE
chore: update templated files

### PR DESCRIPTION
**Pull request by parthea**
- Use the latest version of the post processor which includes updates to the release script dependencies as well as testing for [all protobuf implementations](https://github.com/protocolbuffers/protobuf/tree/main/python#implementation-backends)

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -184,34 +184,48 @@ def default(session):
...
-    session.run("pip", "install", "--upgrade", "pip")
-    session.install("-e", ".[tracing]", "-c", constraints_path)
-    # XXX: Dump installed versions to debug OT issue
-    session.run("pip", "list")
 
-    # Run py.test against the unit tests with OpenTelemetry.
+@nox.session(python=UNIT_TEST_PYTHON_VERSIONS)
+@nox.parametrize(
+    "protobuf_implementation",
+    ["python", "upb", "cpp"],
```
Comment (harshachinta) [inline noxfile.py:191]: @parthea Just curious to understand what does this `protobuf_implementation` mean and why do we have 3 options `["python", "upb", "cpp"]`?

**Comment by harshachinta** [comment]
@parthea Can you please add PR description for this change so that it will help understand for someone looking at this change.

**Review by harshachinta — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (parthea) [reply noxfile.py:191]: This is a feature of `protobuf` where they offer [three separate implementations](https://github.com/protocolbuffers/protobuf/tree/main/python#implementation-backends) of the `Python Protobuf` API. There is an expectation that we should support all 3 implementations in the Cloud Python SDK.

`upb` and `python ` are actively supported in the latest `Protobuf 5.x` while `cpp` is only available in `Protobuf 3.x`

**Comment by parthea** [comment]
> Can you please help add PR description for this change so that it will help understand for someone looking at this change.

Done

> Also what changes will get affected with this?

This only touches the release script dependencies and updates to testing to include all [protobuf implementation backends](https://github.com/protocolbuffers/protobuf/tree/main/python#implementation-backends). There are no user facing changes.

**Pull request merged**